### PR TITLE
Closes #1127: Expose ReloadPageButton through BrowserToolbar

### LINKED ISSUE #1127: Replace refresh button with X to cancel loading during page load
The prototype has this functionality, see custom ReloadPageAction:
* https://github.com/mozilla-mobile/android-components/blob/master/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt#L133-L167
* https://github.com/mozilla-mobile/android-components/blob/master/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt#L277-L290

### LINKED ISSUE #1231: Android Components Sticker Research
We want to make a sticker for a-c that shows describes what we do in a single simple visual design.

## Ideas
1. Use [Material Design launcher icon][3] guidelines.
2. Abstract design - something with a small colour palette.
3. Include the Android logo in there some.

## Investigation
- [Searching for "components"][0] shows puzzle pieces or [circuit components][1].
- [Searching for "modular"][2] shows interconnected boxes.
- Solicit ideas from people with a better sense of design (cc: @brampitoyo @aminalhazwani 👋😅)

[0]: https://duckduckgo.com/?q=components+icon&iar=images&iax=images&ia=images
[1]: https://www.iconfinder.com/icons/381906/circuit_component_current_electronic_microcontrolller_power_technology_icon
[2]: https://duckduckgo.com/?q=modular+icon&iar=images&iax=images&ia=images
[3]: https://material.io/design/iconography/#

### PR DESCRIPTION
**Pull request by jonalmeida**
This change is built off of #1251 so until that PR is merged, reviewing the second commit will be easier.

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt [hunk 1]
```diff
@@ -437,6 +438,42 @@ class BrowserToolbar @JvmOverloads constructor(
         }
     }
 
+    /**
+     * A custom page action that either shows a reload button or a stop button based on the provided
+     * <code>isLoading</code> lambda.
```
Comment (pocmo) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: nit: It's not "custom" anymore. :)

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt [hunk 2]
```diff
@@ -437,6 +438,42 @@ class BrowserToolbar @JvmOverloads constructor(
...
+        private val stopContentDescription: String,
+        private val isLoading: () -> Boolean = { true },
+        background: Int = 0,
+        listener: () -> Unit
+    ) : BrowserToolbar.Button(
+        reloadImage,
+        reloadContentDescription,
+        listener = listener,
+        background = background
+    ) {
```
Comment (pocmo) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: Can we generalize that into a two-state-button? Technically this isn't limited to the "Reload" button anymore. :)

Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: StateChangePageButton? Can do. 👌

Comment (pocmo) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: Similar things in Android: `CompoundButton` or `TwoStatePreference`.

> StateChangePageButton

`Page` probably has no meaning here :)

**Comment by jonalmeida** [comment]
Whoops, referenced the sticker PR here by mistake. 😅

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt (hunk 2 above)
Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: I need a naming intervention.

I stuck with TwoStateButton.  ¯\_(ツ)_/¯

Comment (pocmo) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: Sounds good!

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt [hunk 3]
```diff
@@ -403,36 +406,74 @@ class BrowserToolbar @JvmOverloads constructor(
...
+    /**
+     * An action that either shows an active button or an inactive button based on the provided
+     * <code>isLoading</code> lambda.
+     */
+    open class TwoStateButton(
+        private val enabledImage: Drawable,
+        private val enabledContentDescription: String,
+        private val disabledImage: Drawable,
+        private val disabledContentDescription: String,
+        private val isLoading: () -> Boolean = { true },
```
Comment (pocmo) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: We'd need to find a different name for this lambda too! :)

Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: Oops, fixed!

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt [hunk 4]
```diff
@@ -403,36 +406,74 @@ class BrowserToolbar @JvmOverloads constructor(
...
+        enabledImage,
+        enabledContentDescription,
+        listener = listener,
+        background = background
+    ) {
+        var loading: Boolean = false
+            private set
+
+        override fun bind(view: View) {
+            loading = isEnabled.invoke()
```
Comment (pocmo) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: loading :)

Comment (jonalmeida) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt]: Well that's embarrassing.. 🤦‍♂️

**Pull request merged**